### PR TITLE
Add experimental dark mode flag

### PR DESCRIPTION
**Pull request by rzhao271**
Ref #84453

With the experimental dark mode flag + Electron 16, native titlebars for VS Code on Windows now match with both light mode and dark mode system themes.

Screenshot from custom build:
Example

### CODE REVIEW

File: src/vs/platform/native/electron-main/nativeHostMainService.ts [hunk 1]
```diff
@@ -69,7 +69,6 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 	}
 
 	private registerListeners(): void {
```
Comment (bpasero) [inline src/vs/platform/native/electron-main/nativeHostMainService.ts:71]: Unnecessary change.

File: build/builtin/main.js [hunk 2]
```diff
@@ -31,7 +31,8 @@ app.once('ready', () => {
 			contextIsolation: false,
 			enableWebSQL: false,
 			nativeWindowOpen: true
-		}
+		},
```
Comment (bpasero) [inline build/builtin/main.js:34]: I don't think this window is needed to be dark mode.

File: src/vs/platform/issue/electron-main/issueMainService.ts [hunk 3]
```diff
@@ -335,10 +335,11 @@ export class IssueMainService implements ICommonIssueService {
...
 				zoomFactor: zoomLevelToZoomFactor(options.zoomLevel),
 				sandbox: true,
-				contextIsolation: true,
+				contextIsolation: true
 			},
-			alwaysOnTop: options.alwaysOnTop
-		});
+			alwaysOnTop: options.alwaysOnTop,
+			experimentalDarkMode: true
+		} as any);
```
Comment (bpasero) [inline src/vs/platform/issue/electron-main/issueMainService.ts:342]: Instead, `BrowserWindowOptions & { experimentalDarkmode: boolean }`

File: src/vs/platform/windows/electron-main/window.ts [hunk 4]
```diff
@@ -243,6 +243,8 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 				}
 			}
 
+			(options as any).experimentalDarkMode = true;
```
Comment (bpasero) [inline src/vs/platform/windows/electron-main/window.ts:246]: `BrowserWindowOptions & { experimentalDarkmode: boolean }`

**Review by bpasero — CHANGES_REQUESTED** [review]

**Review by bpasero — APPROVED** [review]
Is this window option temporary or why do we have to enable it explicitly? And what is the impact of the flag specifically?

And can the user override this?

**Review by raghavthind2005 — APPROVED** [review]

**Comment by rzhao271** [comment]
> Is this window option temporary or why do we have to enable it explicitly? And what is the impact of the flag specifically?
> 
> And can the user override this?

Resolved offline.

**Review by deepak1556 — APPROVED** [review]
